torch/nn/qat/modules/linear.py

```diff
@@ -38,17 +40,25 @@ def from_float(cls, mod):
             Args: `mod` a float module, either produced by torch.ao.quantization utilities
             or directly from user
         """
-        assert type(mod) == cls._FLOAT_MODULE, ' qat.' + cls.__name__ + '.from_float only works for ' + \
+        from torch.ao.quantization.utils import nonparam_type  # import error if this is moved to the top
```

Comment:
If anyone knows a good way to avoid the import error or otherwise make this cleaner, let me know.

Reply:
maybe we can just move this util out of `torch.ao` and into `torch.nn.utils.parametrize`?  The function itself is not related to quantization, it's just about parametrizations

Reply:
are we allowed to do that? I am generally wary of messing with other teams' code

Reply:
it sounds reasonable to me in this case, but perhaps we can also ask the folks working on parametrization code?  I see that @albanD  and @soulitzer have committed code to that directory recently, maybe they have some thoughts?

Reply:
I'm not sure what that function is doing. But if it is an API that could be generally useful we're always ok to add it.

To avoid the import issue can you just use `torch.ao.quantization.utils.nonparam_type` here directly?

Reply:
I can it just makes things really hard to read

Reply:
i'll add this and the transfer one into torch.nn.utils.parametrize